x-pack/plugins/security/server/routes/authorization/roles/get_all_by_space.ts

```diff
@@ -53,15 +53,22 @@ export function defineGetAllRolesBySpaceRoutes({
                 logger
               )
             )
-            .filter(
```

Comment:
**note:** This is outside the scope of this PR (we might address it when we integrate the Role Query API), but I just wanted to mention that this code is old and could use some refactoring. We could combine `map` and `filter`, skip transforming if the role is reserved and `hideReservedRoles` is set, avoid iterating through `privilege.feature` if we already know there’s a `base` privilege, and so on 🙂

Reply:
Addressed in 19c1346f3d70a1f5d11fc4126484af56b68b4107

Reply:
Thank you!